upstairs/src/volume.rs

```diff
@@ -684,100 +678,35 @@ impl BlockIO for Volume {
                 let sub_sz = self.block_size as usize
                     * (parent_coverage.end - parent_coverage.start) as usize;
 
-                let parent_buffer = Buffer::new(sub_sz);
-                self.read_only_parent
+                let parent_buffer = self
+                    .read_only_parent
                     .as_ref()
                     .unwrap()
-                    .read(parent_offset, parent_buffer.clone())
+                    .read(parent_offset, Buffer::new(sub_sz))
                     .await?;
 
-                let mut data_vec = data.as_vec().await;
-                let mut owned_vec = data.owned_vec().await;
+                data.eat(data_index, parent_buffer);
```

Comment:
Nit: `eat` could take the buffer by reference, and then we could keep the parent buffer around and try to reuse its allocations in this loop.  We'd need a `Buffer::reset` function or similar to resize + clear, which would reuse the allocations as long as it's not growing.

Reply:
I think 22b8efa covers this, let me know!